rust/lance-encoding/src/encodings/logical/list.rs

```diff
@@ -520,14 +581,32 @@ mod tests {
 
     use std::sync::Arc;
 
+    use arrow_array::builder::{Int32Builder, ListBuilder};
     use arrow_schema::{DataType, Field};
 
-    use crate::testing::check_round_trip_encoding;
+    use crate::testing::{
+        check_round_trip_encoding_of_data, check_round_trip_encoding_random, TestCases,
+    };
 
     #[test_log::test(tokio::test)]
-    async fn test_simple_list() {
+    async fn test_list() {
         let data_type = DataType::List(Arc::new(Field::new("item", DataType::Int32, true)));
         let field = Field::new("", data_type, false);
-        check_round_trip_encoding(field).await;
+        check_round_trip_encoding_random(field).await;
+    }
+
+    #[test_log::test(tokio::test)]
+    async fn test_empty_lists() {
+        // When encoding a list of empty lists there are no items to encode
+        // which is strange and we want to ensure we handle it
+        let items_builder = Int32Builder::new();
+        let mut list_builder = ListBuilder::new(items_builder);
+        list_builder.append(true);
+        list_builder.append(true);
+        list_builder.append(true);
+        let list_array = list_builder.finish();
+
+        let test_cases = TestCases::default().with_range(0..2).with_indices(vec![1]);
+        check_round_trip_encoding_of_data(vec![Arc::new(list_array)], &test_cases).await;
```

Comment:
Maybe we should check various slicing patterns here? That is, something like:

```suggestion
        check_round_trip_encoding_of_data(vec![Arc::new(list_array)], &test_cases).await;
        check_round_trip_encoding_of_data(vec![Arc::new(list_array).slice(3, 0)], &test_cases).await;
        check_round_trip_encoding_of_data(vec![Arc::new(list_array).slice(2, 0), Arc::new(list_array).slice(1, 3)], &test_cases).await;
```

Reply:
It sounds like this might be covered in `check_round_trip_field_encoding_random`. If so, ignore this.

Reply:
Yep :)